spec/lib/guardian_extensions_spec.rb

```diff
@@ -147,6 +147,18 @@
           expect(staff_guardian.can_moderate_chat?(channel.chatable)).to eq(true)
           expect(guardian.can_moderate_chat?(channel.chatable)).to eq(false)
         end
+
+        context "when enable_category_group_moderation is true" do
+          before { SiteSetting.enable_category_group_moderation = true }
+
+          it "returns true if the regular user is part of the reviewable_by_group for the category" do
+            mods = Fabricate(:group)
+            GroupUser.create(user: user, group: mods)
```

Comment:
Perhaps we can just add the user to the group directly? If I am not wrong, we run some special callbacks when adding a user to a group so just manually creating the `GroupUser` record might lead to surprises .

```suggestion
            mods.add(user)
```

Reply:
What do you think about testing this as part of the request spec for chat_controller which I feel will help us to achieve higher coverage since the controller and the guardian is being tested at the same time.

Reply:
Yep definitely to both of these, we already have a `can_moderate` test for the controller that I can add this to :+1: